### PR TITLE
Implement router hierarchy to be able to attach middlewares to different routers

### PR DESCRIPTION
**Pull request by dgaubert**
1 of 3 PRs:
- https://github.com/CartoDB/CartoDB-SQL-API/pull/619
- https://github.com/CartoDB/CartoDB-SQL-API/pull/620
- https://github.com/CartoDB/CartoDB-SQL-API/pull/622

In this first PR, I had to refactor a bit by extracting some functionality with middlewares. Besides, I applied some ES6/7/8 goodies.

For CRers, review the following commits:
 - https://github.com/CartoDB/CartoDB-SQL-API/commit/158f28c8e748a42cddd12934c0acd014ad1a6bc5
 - https://github.com/CartoDB/CartoDB-SQL-API/commit/6c775c1857d554d92ba7d33baa1e5ae41b4593c2
- https://github.com/CartoDB/CartoDB-SQL-API/pull/619/commits/07cf020923bdf54cc87f32d6815a6bc25ed89378

### CODE REVIEW

File: app/controllers/health_check_controller.js [hunk 1]
```diff
@@ -1,35 +1,40 @@
...
+        }
+
+        const startTime = Date.now();
+
+        healthCheckBackend.check((err) => {
+            const ok = !err;
+            const response = {
                 enabled: true,
-                ok: ok,
+                ok,
```
Comment (Algunenano) [inline app/controllers/health_check_controller.js]: I don't understand this change. What are the values of response? Does response.ok exists any more?

File: app/middlewares/cors.js [hunk 2]
```diff
@@ -21,6 +21,10 @@ module.exports = function cors(extraHeaders = []) {
         res.header('Access-Control-Allow-Headers', headers.join(', '));
         res.header('Access-Control-Expose-Headers', exposedHeaders.join(', '));
 
+        if (req.method === 'OPTIONS') {
```
Comment (Algunenano) [inline app/middlewares/cors.js]: You are introducing a different feature here, which is not all that nice to do and doesn't have a NEWS item (in this PR at least).

File: app/server.js [hunk 3]
```diff
@@ -1,69 +1,33 @@
...
-var UserLimitsService = require('./services/user_limits');
-var BatchLogger = require('../batch/batch-logger');
-var JobPublisher = require('../batch/pubsub/job-publisher');
-var JobQueue = require('../batch/job_queue');
-var JobBackend = require('../batch/job_backend');
-var JobCanceller = require('../batch/job_canceller');
-var JobService = require('../batch/job_service');
+const express = require('express');
+const fs = require('fs');
+const mkdirp = require('mkdirp');
```
Comment (Algunenano) [inline app/server.js]: mkdirp seems abandoned. Do you think we should consider changing it?

Comment (Algunenano) [reply app/server.js]: In a different PR that is

File: app/controllers/health_check_controller.js (hunk 1 above)
Comment (dgaubert) [reply app/controllers/health_check_controller.js]: ES6 property shorthand: http://es6-features.org/#PropertyShorthand.

File: app/middlewares/cors.js (hunk 2 above)
Comment (dgaubert) [reply app/middlewares/cors.js]: If I'm not wrong, it should behave in the same way. Having a generic middleware (which is bound for all endpoints) and responds when the method is `options` should be equivalent to defining a `app.options('*', ...)` where basically means for any endpoint being accessed with `options` respond immediately (the cors middleware already took place). 

In any case, I'll double-check it.

File: app/server.js (hunk 3 above)
Comment (dgaubert) [reply app/server.js]: 👌

**Pull request merged**